countries.py

```diff
@@ -0,0 +1,39 @@
+import warnings
+
+
+class Countries:
+    def __init__(self):
+        self.__countries_valid = {"AD", "AR", "AS", "AT", "AU", "AX", "BD", "BE", "BG", "BM",
+                                  "BR", "BY", "CA", "CH", "CO", "CR", "CZ", "DE", "DK", "DO",
+                                  "DZ", "ES", "FI", "FO", "FR", "GB", "GB_full", "GF", "GG",
+                                  "GL", "GP", "GT", "GU", "HR", "HU", "IE", "IM", "IN", "IS",
+                                  "IT", "JE", "JP", "LI", "LK", "LT", "LU", "LV", "MC", "MD",
+                                  "MH", "MK", "MP", "MQ", "MT", "MX", "MY", "NC", "NL", "NO",
+                                  "NZ", "PH", "PK", "PL", "PM", "PR", "PT", "RE", "RO", "RU",
+                                  "SE", "SI", "SJ", "SK", "SM", "TH", "TR", "UA", "US", "UY",
+                                  "VA", "VI", "WF", "YT", "ZA"}
+
+    @property
+    def countries_valid(self):
+        return self.__countries_valid
+
+    def get_clean_country(self, country):
+        country = country.upper()
+        if country == 'AR':
+            warnings.warn('The Argentina data file contains the first 5 positions of the postal code.')
+        if country == 'GB_FULL':
+            return 'GB_full'
+        if country in self.__countries_valid:
+            return country
+        else:
+            raise ValueError(('country={} is not a known country code. '
```

Comment:
Maybe we could actually move this code into init,
```py
class _Country:
	def __init__(self, code):
        # ... process the country code here
        self.name = ..

    def get_download_path(self)
        ...
```
then use it as,
```py
country_obj = _Country(country)
self.country = country_obj.name
download_path = country_obj.get_download_path()
```
as the current class calls `get_clean_country` (and can raise associated warnings) several times.

Reply:
Good idea. Done!